File: server/src/main/java/com/jyinshi/content/entity/Media.java
package com.jyinshi.content.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 媒体信息（content 域）。一部影视/一个条目一行，元数据单独存。
 *
 * <p>身份锚定外部 id：优先 {@link #tmdbId}（采集主源），其次 {@link #doubanId}（补录），
 * 都没有则建「仅标题」条目（{@link #metaSource} = none）。网盘链接不在这里，单独挂在 media_link。
 */
@Data
@TableName("media")
public class Media implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private Integer tmdbId;
    private String doubanId;
    private Integer bangumiId;

    /** movie/tv/anime/variety */
    private String type;

    private String title;
    private String originalTitle;
    private Integer year;

    private String poster;
    private String posterThumb;
    private String backdrop;
    private BigDecimal rating;
    private String overview;

    /** 题材标签，逗号分隔 */
    private String genres;
    /** 国家/地区，逗号分隔 */
    private String country;
    /** 主演，逗号分隔 */
    private String actors;
    /** 导演，逗号分隔 */
    private String directors;

    private String releaseDate;
    private Integer episodeCount;

    /** 季数（不含 Specials），仅剧集类 */
    private Integer seasonCount;
    /** Ended / Returning Series 等 */
    private String seriesStatus;
    /** 是否仍在制作/播出 */
    private Boolean inProduction;
    private String lastAirDate;
    private Integer lastSeasonNumber;
    private Integer lastEpisodeNumber;

    /** 热度（榜单/排序用）= hot_seed + 近 N 天行为分，由热度回写 job 刷新 */
    private Integer hot;

    /** 基线热度：TMDB popularity 种子 / 后台手工设定，行为分在其上叠加 */
    private Integer hotSeed;

    /** 0普通 1精品 2专区（付费分层预留） */
    private Integer tier;

    /** 元数据来源：tmdb/douban/quark/manual/none */
    private String metaSource;

    /** 发布状态：0草稿 1已发布 2下架 */
    private Integer pubStatus;

    /** 1=前台隐藏：搜索/分类/首页/详情均不可见（后台仍可编辑） */
    private Integer searchHidden;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: admin/src/api/media.ts
import { http } from './http'

export interface AdminMedia {
  id: number
  tmdbId?: number
  doubanId?: string
  type?: string
  title?: string
  originalTitle?: string
  year?: number
  poster?: string
  posterThumb?: string
  backdrop?: string
  rating?: number
  overview?: string
  episodeCount?: number
  seasonCount?: number
  seriesStatus?: string
  inProduction?: boolean
  lastAirDate?: string
  lastSeasonNumber?: number
  lastEpisodeNumber?: number
  metaSource?: string
  pubStatus?: number
  hot?: number
  tier?: number
  searchHidden?: number
  updatedAt?: string
}

export interface SeasonVO {
  seasonNumber: number
  name?: string
  episodeCount?: number
  airDate?: string
  poster?: string
  overview?: string
}

export interface MediaDetail {
  media: AdminMedia
  seasons: SeasonVO[]
}

/** 兼容旧版扁平 MediaVO 与新版 MediaDetailVO。 */
export function normalizeMediaDetail(data: unknown): MediaDetail {
  if (data && typeof data === 'object' && 'media' in data) {
    const d = data as MediaDetail
    if (d.media) {
      return { media: d.media, seasons: d.seasons ?? [] }
    }
  }
  return { media: data as AdminMedia, seasons: [] }
}

export interface MediaUpdateBody {
  title?: string
  overview?: string
  poster?: string
  year?: number
  pubStatus?: number
  hot?: number
  tier?: number
  /** 1=关键词搜索不出现 */
  searchHidden?: number
  tmdbId?: number | null
}

export interface ManualMediaBody {
  title: string
  type?: string
  year?: number
  poster?: string
  overview?: string
  publish?: boolean
}

interface ApiPage<T> {
  total: number
  page: number
  size: number
  records: T[]
}

const TYPE_LABEL: Record<string, string> = {
  movie: '电影',
  tv: '剧集',
  anime: '动漫',
  variety: '综艺',
}

const SERIES_TYPES = new Set(['tv', 'anime', 'variety'])

export function typeLabel(type?: string) {
  return TYPE_LABEL[type || ''] || type || '-'
}

export function isSeriesType(type?: string) {
  return SERIES_TYPES.has(type || '')
}

/** 列表/详情缩略图：优先 posterThumb。 */
export function listPosterUrl(m: Pick<AdminMedia, 'poster' | 'posterThumb'>) {
  return m.posterThumb || m.poster || ''
}

/** 剧集季数/连载摘要。 */
export function seriesSummary(m: AdminMedia): string {
  if (!isSeriesType(m.type)) return ''
  if (m.seasonCount == null) return ''
  if (m.seasonCount === 0) return '无季拆分'
  const parts: string[] = [`${m.seasonCount} 季`]
  if (m.episodeCount) parts.push(`全 ${m.episodeCount} 集`)
  if (m.inProduction && m.lastSeasonNumber && m.lastEpisodeNumber) {
    parts.push(`更新至 S${m.lastSeasonNumber}E${m.lastEpisodeNumber}`)
  } else if (m.seriesStatus) {
    parts.push(m.seriesStatus)
  }
  return parts.join(' · ')
}

export function fetchList(page = 1, size = 20, type?: string, q?: string, hidden?: boolean) {
  const qs = new URLSearchParams({ page: String(page), size: String(size) })
  if (type) qs.set('type', type)
  if (q?.trim()) qs.set('q', q.trim())
  if (hidden) qs.set('hidden', 'true')
  return http.get<ApiPage<AdminMedia>>(`/admin/media?${qs}`)
}

export function fetchDetail(id: number) {
  return http.get<MediaDetail>(`/admin/media/${id}`).then(normalizeMediaDetail)
}

export function updateMedia(id: number, body: MediaUpdateBody) {
  return http.put<AdminMedia>(`/admin/media/${id}`, body)
}

export function createManualMedia(body: ManualMediaBody) {
  return http.post<AdminMedia>('/admin/media/manual', body)
}

export function storageStatus() {
  return http.get<{ ready: boolean }>('/admin/media/storage-status')
}

export function publicSiteBase() {
  return (import.meta.env.VITE_PUBLIC_SITE as string) || 'http://localhost:5173'
}
